### PR TITLE
Add PeerDetails page

### PR DESCRIPTION
**Pull request by jarolrod**
Alternative to https://github.com/bitcoin-core/gui-qml/pull/387 which expands to think about a reusable row component, allowing to load any element(s) to the value field, not hardcoding in a height of 21 for a row (but still ensuring it's a minimum of 21), proper icon usage for N/A cases, dummy net indicators in place

TODO:
- Ban buttons
- Hover hints
- Disconnected state instead of flying back to the peers table
- Smart text fixups

Screenshot 2024-02-28 at 3 53 43 AM

Link to github actions build artifacts.

[Build Artifacts]()

### CODE REVIEW

File: src/qml/pages/node/PeerDetails.qml [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+            }
+            Column {
+                width: parent.width
+                bottomPadding: 5
+                KeyValueRow {
+                    key: KeyText { text: qsTr("Direction"); }
+                    value: Row {
+                        IconButton {
+                            iconLocation: details.direction === "Inbound" ? "image://images/arrow-down" : "image://images/arrow-up"
+                            icon.height: 9
```
Comment (johnny9) [inline src/qml/pages/node/PeerDetails.qml:180]:
```suggestion
                            icon.height: 11
```

Add 2 pixels to compensate for the ~20% padding around the icon source. Makes it closer to the figma and also easier to see.

**Review by johnny9 — COMMENTED** [review]
Looks really good. Only thing that stood out to me was the arrow icon was slight too small but that difference makes it harder to see.

File: src/qml/controls/KeyValueRow.qml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+RowLayout {
+    id: root
+    property alias key: key_field.contentItem
+    property alias value: value_field.contentItem
+    width: parent.width
+
+    spacing: 10
+    Pane {
+        id: key_field
```
Comment (johnny9) [inline src/qml/controls/KeyValueRow.qml:18]: QML is always camelCase

File: src/qml/pages/node/NodeSettings.qml [hunk 3]
```diff
@@ -190,6 +190,17 @@ Item {
                 nodeSettingsView.pop()
                 peerTableModel.stopAutoRefresh();
             }
+            onPeerSelected: (peerDetails) => {
+                nodeSettingsView.push(peer_details, {"details": peerDetails})
+            }
+        }
+    }
+    Component {
+        id: peer_details
```
Comment (johnny9) [inline src/qml/pages/node/NodeSettings.qml:199]: QML is camelCase

File: src/qml/pages/node/PeerDetails.qml [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+                    }
+                    value: Loader {
+                        sourceComponent: permissionsRow.isPermissioned ? permissioned : not_permissioned
+                    }
+                    Component {
+                        id: permissioned
+                        ValText { text: permissionsRow.permissionsValue; }
+                    }
+                    Component {
+                        id: not_permissioned
```
Comment (johnny9) [inline src/qml/pages/node/PeerDetails.qml:84]: QML is camelCase

File: src/qml/pages/node/PeerDetails.qml [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+                                icon.color: Theme.color.neutral6
+                            }
+                        }
+                    }
+                }
+                KeyValueRow { key: KeyText { text: qsTr("Version"); } value: ValText { text: details.version; }}
+                KeyValueRow { key: KeyText { text: qsTr("User agent"); } value: ValText { text: details.userAgent; }}
+                KeyValueRow { key: KeyText { text: qsTr("Services"); } value: ValText { text: details.services; }}
+                KeyValueRow {
+                    id: transaction_relay_row
```
Comment (johnny9) [inline src/qml/pages/node/PeerDetails.qml:97]: QML is camelCase

File: src/qml/pages/node/PeerDetails.qml [hunk 6]
```diff
@@ -0,0 +1,286 @@
...
+                horizontalAlignment: Qt.AlignLeft;
+                color: Theme.color.neutral9;
+            }
+
+            Column {
+                width: parent.width
+                bottomPadding: 5
+
+                KeyValueRow { key: KeyText { text: qsTr("Address"); } value: ValText { text: details.address; color: Theme.color.neutral9; }}
+                KeyValueRow { key: KeyText { text: qsTr("VIA"); } value: ValText { text: details.addressLocal; color: Theme.color.neutral9; }}
```
Comment (MarnixCroes) [inline src/qml/pages/node/PeerDetails.qml:66]: maybe _VIA_ could be displayed only when it has a value?
for example, for an onion peer there is no VIA

**Review by MarnixCroes — COMMENTED** [review]
looks good.

onion address can be a bit cut off when not on full window
image

nit, the spacing of _Ping wait_ looks a bit odd
image

File: src/qml/pages/node/PeerDetails.qml (hunk 6 above)
Comment (pablomartin4btc) [reply src/qml/pages/node/PeerDetails.qml:66]: I do agree, unless there's reason for leaving it empty.

Comment (pablomartin4btc) [reply src/qml/pages/node/PeerDetails.qml:66]: As I saw in previous approach of the `PeerDetails` page #387, perhaps a "`N/A`" could be displayed or "`--`"? ([e.g.](https://github.com/bitcoin-core/gui-qml/pull/387#issuecomment-1943106202))

**Review by pablomartin4btc — COMMENTED** [review]
tACK 4909ba591834840cb0238c95a1ceeb8beecb0942

Screenshot from 2024-03-13 12-53-22

**Review by MarnixCroes — APPROVED** [review]
apart from some (minor) things that can be improved indeed, this works as expected.

tACK 02d3121218e1dc04582a85149cc3eabe795aef03

File: src/qml/pages/node/PeerDetails.qml [hunk 7]
```diff
@@ -0,0 +1,286 @@
...
+        property alias text: valText.text
+        spacing: 0
+        NetStatIndicator {}
+        ValText {
+            id: valText
+            anchors.verticalCenter: parent.verticalCenter
+        }
+    }
+}
+
```
Comment (D33r-Gee) [inline src/qml/pages/node/PeerDetails.qml:285]: nit: there's seems to be extra lines at the bottom. Is that on purpose?

**Review by D33r-Gee — COMMENTED** [review]
tACK 02d3121218e1dc04582a85149cc3eabe795aef03 

Other than the nit in the PeersDetails.qml looks good to go, see screenshots below

<details>
  <summary>Ubuntu 22.04</summary>

Screenshot 2024-09-27 095631


</details>

<details>
  <summary>Android Screenshot</summary>

Android_Peers_Detail


</details>

**Review by pablomartin4btc — COMMENTED** [review]
utACK 02d3121218e1dc04582a85149cc3eabe795aef03

Minor changes since my last review. Perhaps an issue could be created with the TODO list.

**Comment by hebasto** [comment]
CI [reports](https://github.com/bitcoin-core/gui-qml/actions/runs/11076549194/job/30779860142) a linker error on macOS 13:
```
ld: Undefined symbols:
  qInitResources_bitcoin_qml(), referenced from:
      QmlGuiMain(int, char**) in libbitcoinqt.a[60](qt_libbitcoinqt_a-bitcoin.o)
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by hebasto** [comment]
> CI [reports](https://github.com/bitcoin-core/gui-qml/actions/runs/11076549194/job/30779860142) a linker error on macOS 13:
> 
> ```
> ld: Undefined symbols:
>   qInitResources_bitcoin_qml(), referenced from:
>       QmlGuiMain(int, char**) in libbitcoinqt.a[60](qt_libbitcoinqt_a-bitcoin.o)
> clang: error: linker command failed with exit code 1 (use -v to see invocation)
> ```

Fixed in https://github.com/bitcoin-core/gui-qml/pull/425.

**Pull request merged**